pkg/oauth/api/v1/types.go

```diff
@@ -86,12 +86,30 @@ type OAuthClient struct {
 	// RedirectURIs is the valid redirection URIs associated with a client
 	RedirectURIs []string `json:"redirectURIs,omitempty"`
 
+	// GrantStrategy determines how to handle grants for this client. If no method is provided, the
+	// cluster default grant handling method will be used. Valid grant handling methods are:
+	//  - auto:   always approves grant requests, useful for trusted clients
+	//  - prompt: prompts the end user for approval of grant requests, useful for third-party clients
+	//  - deny:   always denies grant requests, useful for black-listed clients
+	GrantStrategy GrantHandlerType `json:"grantStrategy,omitempty"`
```

Comment:
this has been controlled by a field named "method" (or "serviceAccountMethod") in the config up until now... let's keep that terminology and use `GrantMethod` (and `grantMethod` in the JSON)